Issues/1988

sbespalov opened the pull request:
# Pull Request Description

This pull request closes #1988

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [ ] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [ ] No

# Code Review And Pre-Merge Checklist

* [ ] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code in hard-to-understand areas.
* [ ] My changes generate no new warnings.
* [ ] I have added tests that prove my fix is effective or that my feature works.
* [ ] New and existing unit tests pass locally with my changes.

sbespalov committed:
issues/1988: fixed `UserHasSecurityRoles` edges creation traversal
within UserEntity unfold

carlspring commented on strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/ConfigurableProviderManager.java:
```diff
@@ -120,6 +124,7 @@ public UserDetails loadUserByUsername(String username)
 
     protected Optional<User> loadExternalUserDetails(String username)
     {
+        logger.info("Load external user derails: username=[{}]", username);
```
Typo: 
```suggestion
        logger.info("Load external user details: username=[{}]", username);
```

carlspring commented on strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/ConfigurableProviderManager.java:
```diff
@@ -134,7 +139,8 @@ public UserDetails loadUserByUsername(String username)
             }
-        
+            
+            logger.info("Fetched external user derails: username=[{}], sourceId=[{}]", externalUser.getUsername(), sourceId);
```
Typo:
```suggestion
            logger.info("Fetched external user details: username=[{}], sourceId=[{}]", externalUser.getUsername(), sourceId);
```

carlspring commented on strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/DatabaseExternalUsersCacheManager.java:
```diff
@@ -26,28 +28,68 @@
             {
-                deleteByUsername(optionalUser.map(u -> user.getUuid()).get());
-                optionalUser = Optional.empty();
+                logger.info("Invalidate user from another source: username=[{}], oldSource=[{}], newSource=[{}]",
```
This should probably be `debug` instead:

sbespalov replied:
fixed

carlspring commented on strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/DatabaseExternalUsersCacheManager.java:
```diff
@@ -59,7 +101,15 @@ public User cacheExternalUserDetails(String sourceId,
 
-            return userRepository.save(userEntry);
+            UserEntity result = userRepository.save(userEntry);
+            logger.info("Cached user: username=[{}], id=[{}], uuid=[{}], sourceId=[{}], lastUpdated=[{}]",
```
...And this:

sbespalov replied:
fixed

carlspring commented on strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/users/service/impl/DatabaseUserService.java:
```diff
@@ -151,6 +153,7 @@ public User save(User user)
 
     public void expireUser(String username, boolean clearSourceId)
     {
+        logger.info("Expire user: username=[{}]", username);
```
Same here.

sbespalov replied:
fixed

carlspring commented on strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/DatabaseExternalUsersCacheManager.java:
```diff
@@ -26,28 +28,68 @@
+            user = new UserData(springUser);
+        }
         String username = user.getUsername();
+        logger.info("Cache external user: username=[{}], id=[{}], uuid=[{}], sourceId=[{}], UserDetails=[{}]",
```
Debug as well.

sbespalov commented:
@carlspring could I merge?

steve-todorov commented:
> @carlspring could I merge?

Please check the Jenkins related issues before merging. Ping me in the chat, if you need me :)

carlspring commented:
> @carlspring could I merge?

We can only merge this, once the tests are passing in Jenkins as well. Could you please look into this?

carlspring commented:
This is failing in relation to #1960.

sbespalov committed:
issues/1988: disable
NugetRemoteRepositoryTest.testRepositoryIndexFetching

sbespalov committed:
issues/1988: disable
NugetArtifactControllerTest.testRemoteProxyGroup and
NugetArtifactControllerTest.testRemoteLastVersion
